R/fread.R

```diff
@@ -226,10 +222,14 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="UTC")
             new_names[ii] %chin% colClasses[[ new_types[ii] ]]
           }))) {
             plural = sum(idx_type) > 1L
-            message('colClasses dictated by user input and those read from YAML header are in conflict (specifically, for column', if (plural) 's',
-                    ' [', paste(new_names[matched_name_idx[!idx_type]], collapse = ','), ']); the proceeding assumes the user input was ',
-                    'an intentional override and will ignore the types implied by the YAML header; please exclude ',
-                    if (plural) 'these columns' else 'this column from colClasses if this was unintentional.')
+            message(domain = NA, sprintf(
+              ngettext(
+                sum(idx_type),
+                'colClasses dictated by user input and those read from YAML header are in conflict (specifically, for column [%s]); the proceeding assumes the user input was an intentional override and will ignore the types implied by the YAML header; please exclude this column from colClasses if this was unintentional.',
+                'colClasses dictated by user input and those read from YAML header are in conflict (specifically, for columns [%s]); the proceeding assumes the user input was an intentional override and will ignore the types implied by the YAML header; please exclude these columns from colClasses if this was unintentional.'
```

Comment:
I think we're safe for now because Chinese doesn't really have pluralization, and I only see maybe extending to a few romance languages (which will have similar pluralization to English), so I agree it's nice to reduce the overhead here given how clunky `ngettext()` is. we can revisit in the future on request.

i also just wrote a message elsewhere that is like `start(s): %d; end(s); %d, which, IINM, would require nested `ngettext()` calls to "properly" pluralize. yuck.